modules/pricer/src/main/java/com/opengamma/strata/pricer/credit/cds/CreditDiscountFactors.java

```diff
@@ -0,0 +1,267 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.pricer.credit.cds;
+
+import java.time.LocalDate;
+import java.util.Optional;
+
+import com.opengamma.strata.basics.currency.Currency;
+import com.opengamma.strata.basics.date.DayCount;
+import com.opengamma.strata.collect.array.DoubleArray;
+import com.opengamma.strata.data.MarketDataName;
+import com.opengamma.strata.market.MarketDataView;
+import com.opengamma.strata.market.curve.Curve;
+import com.opengamma.strata.market.curve.CurveName;
+import com.opengamma.strata.market.param.CurrencyParameterSensitivities;
+import com.opengamma.strata.market.param.CurrencyParameterSensitivity;
+import com.opengamma.strata.market.param.ParameterPerturbation;
+import com.opengamma.strata.market.param.ParameterizedData;
+import com.opengamma.strata.pricer.DiscountFactors;
+import com.opengamma.strata.pricer.ZeroRateSensitivity;
+
+/**
+ * Provides access to discount factors for a single currency.
+ * <p>
+ * The discount factor represents the time value of money for the specified currency
+ * when comparing the valuation date to the specified date.
+ * <p>
+ * This is also used for representing survival probabilities of a legal entity for a single currency.
+ */
+public interface CreditDiscountFactors
+    extends MarketDataView, ParameterizedData {
+
+  /**
+   * Gets the currency.
+   * <p>
+   * The currency that discount factors are provided for.
+   * 
+   * @return the currency
+   */
+  public abstract Currency getCurrency();
+
+  /**
+   * Obtains day count convention.
+   * <p>
+   * This is typically the day count convention of the underlying curve.
+   * 
+   * @return the day count
+   */
+  public abstract DayCount getDayCount();
+
+  /**
+   * Finds the market data structure underlying this instance with the specified name.
+   * <p>
+   * This is most commonly used to find a {@link Curve} using a {@link CurveName}.
+   * If the market data cannot be found, empty is returned.
+   * 
+   * @param <T>  the type of the market data value
+   * @param name  the name to find
+   * @return the market data value, empty if not found
+   */
+  public abstract <T> Optional<T> findData(MarketDataName<T> name);
+
+  /**
+   * Creates an instance of {@link DiscountFactors}.
+   * 
+   * @return the instance
+   */
+  public abstract DiscountFactors toDiscountFactors();
+
+  /**
+   * Obtains the parameter keys of the underlying curve.
+   * 
+   * @return the parameter keys
+   */
+  public abstract DoubleArray getParameterKeys();
```

Comment:
The full set of x values is used when numerical/analytical integration is conducted.